### PR TITLE
EM-5 - Per project metrics

### PR DESCRIPTION
**Pull request by hdamico**
## What does this PR do?

Generates a new folder structure for metric intervals
image
Also, I create `builders` folder which contains `chartkick` with the different data builders for each entity (project and users_project). Now we can build charkick data for differents entities and not only for `users_project`.

And now we are able to create metrics per project.

Jira ticket: [EM-5](https://rootstrap.atlassian.net/browse/EM-5)

### CODE REVIEW

**Comment by hvilloria** [comment]
👏

File: app/services/builders/chartkick/users_project_data.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+module Builders
+  module Chartkick
+    class UsersProjectData < Builders::Chartkick::Base
+      def initialize(project_id, query)
+        @project_id = project_id
+        @query = query
+      end
+
+      def call
+        retrive_users_project.map do |user_project|
```
Comment (mrubi-rootstrap) [inline app/services/builders/chartkick/users_project_data.rb:10]: I think there's a typo in `retrive`

Comment (hdamico) [reply app/services/builders/chartkick/users_project_data.rb:10]: Yep! Thank you!

File: app/services/metrics/group/weekly.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      end
+
+      private
+
+      def interval
+        INTERVAL
+      end
+
+      def value_timestamp
+        (current_time - @number_of_previous.weeks).beginning_of_week..current_time.end_of_week
```
Comment (mrubi-rootstrap) [inline app/services/metrics/group/weekly.rb:18]: `@number_of_previous.weeks` awesome 👏

Comment (hdamico) [reply app/services/metrics/group/weekly.rb:18]: Credits to @hvilloria!, I just moved the code.

**Pull request merged**